warehouse/ingest-core/src/main/java/datawave/ingest/table/aggregator/GlobalIndexUidAggregator.java

```diff
@@ -170,17 +168,29 @@ public Value reduce(Key key, Iterator<Value> iter) {
                     }
                     
                     for (String uid : v.getUIDList()) {
-                        
                         // check that a removal has not occurred
                         // if it has, we decrement the count, from above.
                         if (!uidsToRemove.contains(uid) && !quarantinedIds.contains(uid)) {
                             
                             // add the UID iff we are under our MAX
                             if (uids.size() < maxUids)
                                 uids.add(uid);
-                            
                         }
-                        
+                    }
+                    
+                    // It's possible due to previous compactions that we have a positive count as well as UIDs in the removed list.
+                    // Add UIDs that aren't already in the UIDs list (e.g, the UID was added back and we saw it in an earlier
+                    // key (which is a newer key with a larger timestamp value).
+                    if (!seenIgnore) {
+                        if (!processInTimestampOrder) {
+                            uidsToRemove.addAll(v.getREMOVEDUIDList());
+                        } else {
+                            for (String uid : v.getREMOVEDUIDList()) {
+                                if (!uids.contains(uid)) {
+                                    uidsToRemove.add(uid);
+                                }
+                            }
+                        }
                     }
```

Comment:
It's really line 186 above that changes existing behavior. Otherwise, the iterator should behave exactly the same with my changes as long as the processInTimestampOrder option isn't passed. Removing this line breaks the new testRemoveAndReAddUUID and testPropagateRemoveAndAdd tests I added, but if it makes more sense, I can change this to the suggestion.

```suggestion
                    if (!seenIgnore && processInTimestampOrder) {
                        for (String uid : v.getREMOVEDUIDList()) {
                            if (!uids.contains(uid)) {
                                uidsToRemove.add(uid);
                            }
                        }
                    }
```